applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_client.c

```diff
@@ -1516,13 +1391,23 @@ void unicast_client_conn_disconnected(struct bt_conn *conn)
 int unicast_client_discover(struct bt_conn *conn, enum unicast_discover_dir dir)
 {
 	int ret;
+	uint8_t index;
 
-	ret = discover_list_add(conn, dir);
+	ret = channel_index_free_get(&index);
```

Comment:
Nitpick: We could add a log in here for showing the device has been set before.

Reply:
We don't check if the device is already in the headsets-array, but I can create one and then return `-EALREADY` if it is there.

Reply:
After check again, I think return is OK too, since the caller will capture it.